stream_alert/rule_processor/rules_engine.py

```diff
@@ -250,6 +328,11 @@ def process(cls, input_payload):
                 if not matcher_result:
                     continue
 
+                types_result = cls.match_types(record,
+                                               payload.normalized_types,
+                                               rule.datatypes)
+                record.update({'normalized_types': types_result})
```

Comment:
You don't have to use `.update` here. `update` is useful if you have a preexisting dictionary that you want to 'append' to another dict, but since you're only adding one key/value item to the dictionary just use:

`record['normalized_types'] = types_result`